src/bp-integrations/learndash/library/SyncGenerator.php

```diff
@@ -753,4 +753,57 @@ protected function getLastSyncTimestamp( $type = 'bp' ) {
 
 		return $lastSync;
 	}
+
+	/**
+	 * Update a ld group based on current bp group
+	 *
+	 * @param string $ld_group_id Group id for learndash
+	 *
+	 * @param int $groupId Group id for buddyboss
+	 *
+	 * @since BuddyBoss 1.5.8
+	 */
+	public function updateLearndashGroup( $ld_group_id, $groupId ) {
+		$bpGroup = groups_get_group( $groupId );
+
+		wp_update_post(
+			array(
+				'ID'           => $ld_group_id,
+				'post_title'   => $bpGroup->name,
+				'post_author'  => $bpGroup->creator_id,
+				'post_content' => $bpGroup->description,
+				'post_status'  => 'publish',
+				'post_type'    => learndash_get_post_type_slug( 'group' ),
+			)
+		);
+	}
+
+	/**
+	 * Update a bp group based on current ld group
+	 *
+	 * @param string $ld_group_id Group id for learndash
+	 *
+	 * @param int $groupId Group id for buddyboss
+	 *
+	 * @since BuddyBoss 1.5.8
+	 */
+	public function updateBuddypressGroup( $ld_group_id, $groupId ) {
+		$ldGroup  = get_post( $ld_group_id );
+		$settings = bp_ld_sync( 'settings' );
+
+		groups_create_group(
+			array(
+				'group_id'    => $groupId,
+				'creator_id'  => $ldGroup->post_author,
+				'name'        => $ldGroup->post_title ?: "For Social Group: {$this->ldGroupId}",
+				'status'      => $settings->get( 'learndash.default_bp_privacy' ),
+				'description' => $ldGroup->post_content,
+				'slug'        => $ldGroup->post_name,
+			)
+		);
```

Comment:
@jitendrabanjara1991
```suggestion
if ( ! empty ( $groupId ) ) {
		groups_create_group(
			array(
				'group_id'    => $groupId,
				'creator_id'  => $ldGroup->post_author,
				'name'        => $ldGroup->post_title ?: "For Social Group: {$this->ldGroupId}",
				'status'      => $settings->get( 'learndash.default_bp_privacy' ),
				'description' => $ldGroup->post_content,
				'slug'        => $ldGroup->post_name,
			)
		);
		}
```